status/condition_set.go

```diff
@@ -237,3 +238,27 @@ func (c ConditionSet) findUnhealthyDependents() []Condition {
 	})
 	return conditions
 }
+
+// GetConditionForUnstructured gets the condition for the given conditionPath. It returns an error if the condition is not found.
+func GetConditionForUnstructured(conditionPath string, nodeClass *unstructured.Unstructured) (*Condition, error) {
```

Comment:
It looks like the majority of this code is converting between a runtime.Object and a status.Object. WDYT about instead making a helper like:

```
package status

func ObjectFrom(runtime.Object) {
   ...
}
```

And then we can reuse the standard condition handling methods?

Reply:
Agreed. Added that change.